<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1640</width>
    <height>1150</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>1640</width>
    <height>1150</height>
   </size>
  </property>
  <property name="maximumSize">
   <size>
    <width>1640</width>
    <height>1150</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QGroupBox" name="groupBox_info">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>40</y>
      <width>981</width>
      <height>61</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>12</pointsize>
     </font>
    </property>
    <property name="title">
     <string>Информация о проекте</string>
    </property>
    <widget class="QWidget" name="layoutWidget">
     <property name="geometry">
      <rect>
       <x>170</x>
       <y>30</y>
       <width>641</width>
       <height>23</height>
      </rect>
     </property>
     <layout class="QHBoxLayout" name="horizontalLayout">
      <item>
       <widget class="QLabel" name="label_vertices">
        <property name="text">
         <string>Количество вершин</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="label_verticesValue">
        <property name="text">
         <string/>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="label_edges">
        <property name="text">
         <string>Количество рёбер</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="label_edgesValue">
        <property name="text">
         <string/>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </widget>
   <widget class="QLineEdit" name="lineEdit_filename">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>14</y>
      <width>640</width>
      <height>21</height>
     </rect>
    </property>
    <property name="focusPolicy">
     <enum>Qt::NoFocus</enum>
    </property>
   </widget>
   <widget class="QPushButton" name="pushButton_open">
    <property name="geometry">
     <rect>
      <x>680</x>
      <y>10</y>
      <width>151</width>
      <height>32</height>
     </rect>
    </property>
    <property name="focusPolicy">
     <enum>Qt::TabFocus</enum>
    </property>
    <property name="text">
     <string>Открыть файл</string>
    </property>
   </widget>
   <widget class="QPushButton" name="pushButton_saveImage">
    <property name="geometry">
     <rect>
      <x>1470</x>
      <y>650</y>
      <width>151</width>
      <height>32</height>
     </rect>
    </property>
    <property name="text">
     <string>Сделать screenshot</string>
    </property>
   </widget>
   <widget class="QPushButton" name="pushButton_makeGIF">
    <property name="geometry">
     <rect>
      <x>1470</x>
      <y>610</y>
      <width>151</width>
      <height>32</height>
     </rect>
    </property>
    <property name="text">
     <string>Записать GIF</string>
    </property>
   </widget>
   <widget class="QPushButton" name="pushButton_clean">
    <property name="geometry">
     <rect>
      <x>850</x>
      <y>10</y>
      <width>151</width>
      <height>32</height>
     </rect>
    </property>
    <property name="text">
     <string>Очистить</string>
    </property>
   </widget>
   <widget class="QGroupBox" name="groupBox_projection">
    <property name="geometry">
     <rect>
      <x>1470</x>
      <y>110</y>
      <width>151</width>
      <height>51</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>12</pointsize>
     </font>
    </property>
    <property name="title">
     <string>Тип проекции</string>
    </property>
    <widget class="QComboBox" name="comboBox_projectionType">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>20</y>
       <width>151</width>
       <height>32</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>13</pointsize>
      </font>
     </property>
     <item>
      <property name="text">
       <string>параллельная</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>центральная</string>
      </property>
     </item>
    </widget>
   </widget>
   <widget class="QGroupBox" name="groupBox_edges">
    <property name="geometry">
     <rect>
      <x>1470</x>
      <y>210</y>
      <width>151</width>
      <height>171</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>12</pointsize>
     </font>
    </property>
    <property name="title">
     <string>Рёбра</string>
    </property>
    <widget class="QLabel" name="label_edgeThickness">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>80</y>
       <width>131</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>Толщина линии</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_edgeType">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>30</y>
       <width>131</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>Тип линии</string>
     </property>
    </widget>
    <widget class="QSpinBox" name="spinBox_edgeThickness">
     <property name="geometry">
      <rect>
       <x>5</x>
       <y>100</y>
       <width>140</width>
       <height>22</height>
      </rect>
     </property>
     <property name="focusPolicy">
      <enum>Qt::ClickFocus</enum>
     </property>
     <property name="alignment">
      <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
     </property>
     <property name="minimum">
      <number>1</number>
     </property>
     <property name="maximum">
      <number>99</number>
     </property>
    </widget>
    <widget class="QComboBox" name="comboBox_edgeType">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>50</y>
       <width>151</width>
       <height>32</height>
      </rect>
     </property>
     <item>
      <property name="text">
       <string>сплошная</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>пунктирная</string>
      </property>
     </item>
    </widget>
    <widget class="QPushButton" name="pushButton_edgeColor">
     <property name="geometry">
      <rect>
       <x>5</x>
       <y>130</y>
       <width>140</width>
       <height>32</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>13</pointsize>
      </font>
     </property>
     <property name="text">
      <string>Цвет линии</string>
     </property>
    </widget>
   </widget>
   <widget class="QGroupBox" name="groupBox_vertices">
    <property name="geometry">
     <rect>
      <x>1470</x>
      <y>390</y>
      <width>151</width>
      <height>171</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>12</pointsize>
     </font>
    </property>
    <property name="title">
     <string>Вершины</string>
    </property>
    <widget class="QSpinBox" name="spinBox_verticeSize">
     <property name="geometry">
      <rect>
       <x>5</x>
       <y>100</y>
       <width>140</width>
       <height>22</height>
      </rect>
     </property>
     <property name="focusPolicy">
      <enum>Qt::ClickFocus</enum>
     </property>
     <property name="alignment">
      <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
     </property>
     <property name="minimum">
      <number>1</number>
     </property>
     <property name="value">
      <number>1</number>
     </property>
    </widget>
    <widget class="QComboBox" name="comboBox_verticeType">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>50</y>
       <width>151</width>
       <height>32</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>13</pointsize>
      </font>
     </property>
     <item>
      <property name="text">
       <string>отсутствует</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>круг</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>квадрат</string>
      </property>
     </item>
    </widget>
    <widget class="QLabel" name="label_verticeMethod">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>30</y>
       <width>131</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>Способ отображения</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_verticeSize">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>80</y>
       <width>131</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>Размер вершины</string>
     </property>
    </widget>
    <widget class="QPushButton" name="pushBotton_verticeColor">
     <property name="geometry">
      <rect>
       <x>5</x>
       <y>130</y>
       <width>140</width>
       <height>32</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>13</pointsize>
      </font>
     </property>
     <property name="text">
      <string>Цвет вершины</string>
     </property>
    </widget>
   </widget>
   <widget class="QPushButton" name="pushButton_resetSettings">
    <property name="geometry">
     <rect>
      <x>1470</x>
      <y>570</y>
      <width>151</width>
      <height>32</height>
     </rect>
    </property>
    <property name="text">
     <string>Сбросить настройки</string>
    </property>
   </widget>
   <widget class="QGroupBox" name="groupBox_rotate">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>110</y>
      <width>151</width>
      <height>241</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>12</pointsize>
     </font>
    </property>
    <property name="title">
     <string>Повернуть</string>
    </property>
    <widget class="QScrollBar" name="horizontalScrollBar_zRotate">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>210</y>
       <width>131</width>
       <height>16</height>
      </rect>
     </property>
     <property name="minimum">
      <number>-360</number>
     </property>
     <property name="maximum">
      <number>360</number>
     </property>
     <property name="pageStep">
      <number>1</number>
     </property>
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="invertedControls">
      <bool>true</bool>
     </property>
    </widget>
    <widget class="QLabel" name="label_yRotate">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>110</y>
       <width>21</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>y</string>
     </property>
    </widget>
    <widget class="QScrollBar" name="horizontalScrollBar_xRotate">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>70</y>
       <width>131</width>
       <height>16</height>
      </rect>
     </property>
     <property name="minimum">
      <number>-360</number>
     </property>
     <property name="maximum">
      <number>360</number>
     </property>
     <property name="pageStep">
      <number>1</number>
     </property>
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="invertedControls">
      <bool>true</bool>
     </property>
    </widget>
    <widget class="QLabel" name="label_zRotate">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>180</y>
       <width>21</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>z</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_xRotate">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>40</y>
       <width>21</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>x</string>
     </property>
    </widget>
    <widget class="QSpinBox" name="spinBox_xRotate">
     <property name="geometry">
      <rect>
       <x>40</x>
       <y>40</y>
       <width>101</width>
       <height>22</height>
      </rect>
     </property>
     <property name="focusPolicy">
      <enum>Qt::ClickFocus</enum>
     </property>
     <property name="alignment">
      <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
     </property>
     <property name="accelerated">
      <bool>true</bool>
     </property>
     <property name="minimum">
      <number>-360</number>
     </property>
     <property name="maximum">
      <number>360</number>
     </property>
    </widget>
    <widget class="QScrollBar" name="horizontalScrollBar_yRotate">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>140</y>
       <width>131</width>
       <height>16</height>
      </rect>
     </property>
     <property name="minimum">
      <number>-360</number>
     </property>
     <property name="maximum">
      <number>360</number>
     </property>
     <property name="pageStep">
      <number>1</number>
     </property>
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="invertedControls">
      <bool>true</bool>
     </property>
    </widget>
    <widget class="QSpinBox" name="spinBox_yRotate">
     <property name="geometry">
      <rect>
       <x>40</x>
       <y>110</y>
       <width>101</width>
       <height>22</height>
      </rect>
     </property>
     <property name="focusPolicy">
      <enum>Qt::ClickFocus</enum>
     </property>
     <property name="alignment">
      <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
     </property>
     <property name="accelerated">
      <bool>true</bool>
     </property>
     <property name="minimum">
      <number>-360</number>
     </property>
     <property name="maximum">
      <number>360</number>
     </property>
    </widget>
    <widget class="QSpinBox" name="spinBox_zRotate">
     <property name="geometry">
      <rect>
       <x>40</x>
       <y>180</y>
       <width>101</width>
       <height>22</height>
      </rect>
     </property>
     <property name="focusPolicy">
      <enum>Qt::ClickFocus</enum>
     </property>
     <property name="alignment">
      <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
     </property>
     <property name="accelerated">
      <bool>true</bool>
     </property>
     <property name="minimum">
      <number>-360</number>
     </property>
     <property name="maximum">
      <number>360</number>
     </property>
    </widget>
   </widget>
   <widget class="view3d::MyOpenGLWidget" name="glWidget" native="true">
    <property name="geometry">
     <rect>
      <x>190</x>
      <y>120</y>
      <width>1260</width>
      <height>960</height>
     </rect>
    </property>
    <property name="mouseTracking">
     <bool>true</bool>
    </property>
    <widget class="QSlider" name="scale_slider">
     <property name="geometry">
      <rect>
       <x>950</x>
       <y>20</y>
       <width>300</width>
       <height>25</height>
      </rect>
     </property>
     <property name="minimum">
      <number>1</number>
     </property>
     <property name="maximum">
      <number>500</number>
     </property>
     <property name="value">
      <number>100</number>
     </property>
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
    </widget>
   </widget>
   <widget class="QGroupBox" name="groupBox_translate">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>360</y>
      <width>151</width>
      <height>241</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>12</pointsize>
     </font>
    </property>
    <property name="title">
     <string>Переместить</string>
    </property>
    <widget class="QScrollBar" name="horizontalScrollBar_zTranslate">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>210</y>
       <width>131</width>
       <height>16</height>
      </rect>
     </property>
     <property name="minimum">
      <number>-99</number>
     </property>
     <property name="maximum">
      <number>99</number>
     </property>
     <property name="pageStep">
      <number>1</number>
     </property>
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="invertedControls">
      <bool>true</bool>
     </property>
    </widget>
    <widget class="QLabel" name="label_yTranslate">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>110</y>
       <width>21</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>y</string>
     </property>
    </widget>
    <widget class="QSpinBox" name="spinBox_xTranslate">
     <property name="geometry">
      <rect>
       <x>40</x>
       <y>40</y>
       <width>101</width>
       <height>22</height>
      </rect>
     </property>
     <property name="focusPolicy">
      <enum>Qt::ClickFocus</enum>
     </property>
     <property name="alignment">
      <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
     </property>
     <property name="accelerated">
      <bool>true</bool>
     </property>
     <property name="minimum">
      <number>-99</number>
     </property>
     <property name="singleStep">
      <number>1</number>
     </property>
    </widget>
    <widget class="QLabel" name="label_zTranslate">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>180</y>
       <width>21</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>z</string>
     </property>
    </widget>
    <widget class="QSpinBox" name="spinBox_yTranslate">
     <property name="geometry">
      <rect>
       <x>40</x>
       <y>110</y>
       <width>101</width>
       <height>22</height>
      </rect>
     </property>
     <property name="focusPolicy">
      <enum>Qt::ClickFocus</enum>
     </property>
     <property name="alignment">
      <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
     </property>
     <property name="accelerated">
      <bool>true</bool>
     </property>
     <property name="minimum">
      <number>-99</number>
     </property>
    </widget>
    <widget class="QSpinBox" name="spinBox_zTranslate">
     <property name="geometry">
      <rect>
       <x>40</x>
       <y>180</y>
       <width>101</width>
       <height>22</height>
      </rect>
     </property>
     <property name="focusPolicy">
      <enum>Qt::ClickFocus</enum>
     </property>
     <property name="alignment">
      <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
     </property>
     <property name="accelerated">
      <bool>true</bool>
     </property>
     <property name="minimum">
      <number>-99</number>
     </property>
    </widget>
    <widget class="QScrollBar" name="horizontalScrollBar_xTranslate">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>70</y>
       <width>131</width>
       <height>16</height>
      </rect>
     </property>
     <property name="minimum">
      <number>-99</number>
     </property>
     <property name="maximum">
      <number>99</number>
     </property>
     <property name="pageStep">
      <number>1</number>
     </property>
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="invertedControls">
      <bool>true</bool>
     </property>
    </widget>
    <widget class="QLabel" name="label_xTranslate">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>40</y>
       <width>21</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>x</string>
     </property>
    </widget>
    <widget class="QScrollBar" name="horizontalScrollBar_yTranslate">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>140</y>
       <width>131</width>
       <height>16</height>
      </rect>
     </property>
     <property name="minimum">
      <number>-99</number>
     </property>
     <property name="maximum">
      <number>99</number>
     </property>
     <property name="pageStep">
      <number>1</number>
     </property>
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="invertedControls">
      <bool>true</bool>
     </property>
    </widget>
   </widget>
   <widget class="QPushButton" name="pushButton_resetPosition">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>610</y>
      <width>151</width>
      <height>32</height>
     </rect>
    </property>
    <property name="text">
     <string>Сбросить позицию</string>
    </property>
   </widget>
   <widget class="QPushButton" name="pushButton_backgroundColor">
    <property name="geometry">
     <rect>
      <x>1475</x>
      <y>170</y>
      <width>141</width>
      <height>32</height>
     </rect>
    </property>
    <property name="text">
     <string>Цвет фона</string>
    </property>
   </widget>
   <zorder>glWidget</zorder>
   <zorder>groupBox_info</zorder>
   <zorder>lineEdit_filename</zorder>
   <zorder>pushButton_open</zorder>
   <zorder>pushButton_saveImage</zorder>
   <zorder>pushButton_makeGIF</zorder>
   <zorder>pushButton_clean</zorder>
   <zorder>groupBox_projection</zorder>
   <zorder>groupBox_edges</zorder>
   <zorder>groupBox_vertices</zorder>
   <zorder>pushButton_resetSettings</zorder>
   <zorder>groupBox_rotate</zorder>
   <zorder>groupBox_translate</zorder>
   <zorder>pushButton_resetPosition</zorder>
   <zorder>pushButton_backgroundColor</zorder>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <action name="action">
   <property name="text">
    <string>Открыть файл</string>
   </property>
  </action>
  <action name="action_3">
   <property name="text">
    <string>Закрыть все файлы</string>
   </property>
  </action>
  <action name="action_4">
   <property name="text">
    <string>Сообщить об ошибке</string>
   </property>
  </action>
  <action name="action_5">
   <property name="text">
    <string>О программе</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>view3d::MyOpenGLWidget</class>
   <extends>QWidget</extends>
   <header location="global">view/myopenglwidget.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
